pkg/armrpc/asyncoperation/worker/registry.go

```diff
@@ -74,41 +71,40 @@ func (h *ControllerRegistry) Register(ctx context.Context, resourceType string,
 //
 // The default controller will be used when Get is called with an operation type that has no registered controller.
 // The default controller will not be cached by the registry.
-func (h *ControllerRegistry) RegisterDefault(ctx context.Context, factoryFn ControllerFactoryFunc, opts ctrl.Options) error {
+func (h *ControllerRegistry) RegisterDefault(factoryFn ControllerFactoryFunc, opts ctrl.Options) error {
 	h.ctrlMapMu.Lock()
 	defer h.ctrlMapMu.Unlock()
 
+	// Note: we can't call opts.Validate() here because we don't know the resource type yet.
+	if opts.StorageClient == nil {
+		return fmt.Errorf("invalid controller options: .StorageClient is required")
```

Comment:
```suggestion
		return fmt.Errorf("invalid controller options: StorageClient is required")
```

Reply:
I did it this because it's consistent with the other error messages in validate. It has a leading period because go doesn't like error messages that begin with a capital letter.